exercises/rail-fence-cipher/src/example.c

```diff
@@ -0,0 +1,54 @@
+#include "rail_fence_cipher.h"
+#include <string.h>
+#include <stdlib.h>
+
+char *encode(char *text, size_t rails)
+{
+   size_t length = strlen(text);
+   char *ciphertext = (char *)calloc(length + 1, sizeof(char));
+   size_t cell_size = 2 * rails - 2;
+   if (cell_size == 0) {
+      strcpy(ciphertext, text);
+      return ciphertext;
+   }
+   size_t j = 0;
+   for (size_t rail = 0; rail < rails; rail++) {
+      for (size_t i = 0; i < length; i += cell_size) {
+         if (i + rail < length)
+            ciphertext[j++] = text[i + rail];
+         if (rail > 0 && rail < rails - 1) {
+            if (i + rails + rail < length)
+               ciphertext[j++] = text[i + cell_size - rail];
+         }
+      }
+   }
+   return ciphertext;
+}
+
+char *decode(char *ciphertext, size_t rails)
+{
+   size_t length = strlen(ciphertext);
+   char *text = (char *)calloc(length + 1, sizeof(char));
```

Comment:
```suggestion
   char *text = calloc(length + 1, sizeof(char));
```

https://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc